### PR TITLE
Add a wrapper to solve ExaPF problems with MOI compatible solvers

### PR DESCRIPTION
**Pull request by frapac**
This PR brings also some enhancements in the API of `AbstractNLPEvaluators` objects

### CODE REVIEW

File: scripts/moi_solver.jl [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
```
Comment (michel2323) [inline scripts/moi_solver.jl:1]: Shouldn't this maybe be a test?

File: scripts/moi_solver.jl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    MOI.set(optimizer, MOI.NLPBlock(), block_data)
+    MOI.set(optimizer, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+    MOI.optimize!(optimizer)
+end
+
+datafile = joinpath(dirname(@__FILE__), "..", "data", "case57.m")
+
+nlp = ExaPF.ReducedSpaceEvaluator(datafile)
+aug = ExaPF.AugLagEvaluator(nlp, ExaPF.initial(nlp); c₀=0.1, scale=true)
+# optimizer = KNITRO.Optimizer(hessopt=3)
```
Comment (michel2323) [inline scripts/moi_solver.jl:39]: In test you could also add KNITRO.

**Review by michel2323 — CHANGES_REQUESTED** [review]

File: scripts/moi_solver.jl (hunk 1 above)
Comment (frapac) [reply scripts/moi_solver.jl:1]: good point!

File: scripts/moi_solver.jl (hunk 2 above)
Comment (frapac) [reply scripts/moi_solver.jl:39]: I prefer not to, as it is a commercial solver...

**Review by dmaldona — APPROVED** [review]

File: src/Evaluators/MOI_wrapper.jl [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# MOI wrapper
+
+mutable struct ExaEvaluator <: MOI.AbstractNLPEvaluator
```
Comment (michel2323) [inline src/Evaluators/MOI_wrapper.jl:3]: That sounds now totally picky and maybe unnecessary, but you think we shouldn't call it `ExaPFEvaluator` or maybe even just `Evaluator` like in JuMP?

**Review by michel2323 — APPROVED** [review]

File: src/Evaluators/MOI_wrapper.jl (hunk 3 above)
Comment (frapac) [reply src/Evaluators/MOI_wrapper.jl:3]: No, this is particularly relevant. Indeed, I should be more careful with naming. I have renamed it `MOIEvaluator`, to emphasis the fact that we rely on MOI there.

**Pull request closed without merging**